src/github_runner_manager/reactive/process_manager.py

```diff
@@ -0,0 +1,140 @@
+from pathlib import Path
+
+from github_runner_manager.reactive.types_ import RunnerConfig
+from github_runner_manager.utilities import secure_run_subprocess
+
+logger = logging.getLogger(__name__)
+
+REACTIVE_RUNNER_LOG_DIR = Path("/var/log/reactive_runner")
+
+PYTHON_BIN = "/usr/bin/python3"
+REACTIVE_RUNNER_SCRIPT_MODULE = "github_runner_manager.reactive.runner"
+REACTIVE_RUNNER_CMD_LINE_PREFIX = f"{PYTHON_BIN} -m {REACTIVE_RUNNER_SCRIPT_MODULE}"
+PID_CMD_COLUMN_WIDTH = len(REACTIVE_RUNNER_CMD_LINE_PREFIX)
+PIDS_COMMAND_LINE = [
+    "ps",
+    "axo",
+    f"cmd:{PID_CMD_COLUMN_WIDTH},pid",
+    "--no-headers",
+    "--sort=-start_time",
+]
+UBUNTU_USER = "ubuntu"
+RUNNER_CONFIG_ENV_VAR = "RUNNER_CONFIG"
+
+
+class ReactiveRunnerError(Exception):
+    """Raised when a reactive runner error occurs."""
+
+
+def reconcile(quantity: int, runner_config: RunnerConfig) -> int:
+    """Reconcile the number of reactive runner processes.
+
+    Args:
+        quantity: The number of processes to spawn.
+        runner_config: The reactive runner configuration.
+
+    Raises a ReactiveRunnerError if the runner fails to spawn.
+
+    Returns:
+        The number of reactive runner processes spawned/killed.
+    """
+    pids = _get_pids()
+    current_quantity = len(pids)
+    logger.info("Current quantity of reactive runner processes: %s", current_quantity)
+    delta = quantity - current_quantity
+    if delta > 0:
+        logger.info("Will spawn %d new reactive runner process(es)", delta)
+        _setup_logging_for_processes()
+        for _ in range(delta):
+            _spawn_runner(runner_config)
+    elif delta < 0:
+        logger.info("Will kill %d process(es).", -delta)
+        for pid in pids[:-delta]:
+            logger.info("Killing reactive runner process with pid %s", pid)
+            try:
+                os.kill(pid, signal.SIGTERM)
+            except ProcessLookupError:
+                # There can be a race condition that the process has already terminated.
+                # We just ignore and log the fact.
+                logger.info(
+                    "Failed to kill process with pid %s. Process might have terminated it self.",
+                    pid,
+                )
+    else:
+        logger.info("No changes to number of reactive runner processes needed.")
+
+    return delta
+
+
+def _get_pids() -> list[int]:
+    """Get the PIDs of the reactive runners processes.
+
+    Returns:
+        The PIDs of the reactive runner processes sorted by start time in descending order.
+
+    Raises:
+        ReactiveRunnerError: If the command to get the PIDs fails
+    """
+    result = secure_run_subprocess(cmd=PIDS_COMMAND_LINE)
+    if result.returncode != 0:
+        raise ReactiveRunnerError("Failed to get list of processes")
+
+    return [
+        int(line.rstrip().rsplit(maxsplit=1)[-1])
+        for line in result.stdout.decode().split("\n")
+        if line.startswith(REACTIVE_RUNNER_CMD_LINE_PREFIX)
```

Comment:
Nit:
Would it be helpful to have an example stdout to help understand what this processing does?